src/WebJobs.Extensions.DurableTask/Listener/DurableTaskScaleMonitor.cs

```diff
@@ -0,0 +1,180 @@
+    {
+        private readonly string functionId;
+        private readonly FunctionName functionName;
+        private readonly string hubName;
+        private readonly string storageConnectionString;
+        private readonly EndToEndTraceHelper traceHelper;
+        private readonly ScaleMonitorDescriptor scaleMonitorDescriptor;
+
+        private DisconnectedPerformanceMonitor performanceMonitor;
+
+        public DurableTaskScaleMonitor(
+            string functionId,
+            FunctionName functionName,
+            string hubName,
+            string storageConnectionString,
+            EndToEndTraceHelper traceHelper,
+            DisconnectedPerformanceMonitor performanceMonitor = null)
+        {
+            this.functionId = functionId;
+            this.functionName = functionName;
+            this.hubName = hubName;
+            this.storageConnectionString = storageConnectionString;
+            this.performanceMonitor = performanceMonitor;
+            this.traceHelper = traceHelper;
+            this.scaleMonitorDescriptor = new ScaleMonitorDescriptor($"{this.functionId}-DurableTaskTrigger-{this.hubName}".ToLower());
+        }
+
+        public ScaleMonitorDescriptor Descriptor
+        {
+            get
+            {
+                return this.scaleMonitorDescriptor;
+            }
+        }
+
+        private DisconnectedPerformanceMonitor PerformanceMonitor
+        {
+            get
+            {
+                if (this.performanceMonitor == null)
+                {
+                    this.performanceMonitor = new DisconnectedPerformanceMonitor(this.storageConnectionString, this.hubName);
+                }
+
+                return this.performanceMonitor;
+            }
+        }
+
+        async Task<ScaleMetrics> IScaleMonitor.GetMetricsAsync()
+        {
+            return await this.GetMetricsAsync();
+        }
+
+        public async Task<DurableTaskTriggerMetrics> GetMetricsAsync()
+        {
+            DurableTaskTriggerMetrics metrics = new DurableTaskTriggerMetrics();
+
+            // Durable stores its own metrics, so we just collect them here
+            PerformanceHeartbeat heartbeat = null;
+            try
+            {
+                heartbeat = await this.PerformanceMonitor.PulseAsync();
+            }
+            catch (StorageException e)
+            {
+                string errorMsg = string.Empty;
+                var webException = e.InnerException as WebException;
+                if (webException != null && webException.Status == WebExceptionStatus.ProtocolError)
+                {
+                    string statusCode = ((HttpWebResponse)webException.Response)?.StatusCode.ToString();
+                    string statusDesc = ((HttpWebResponse)webException.Response)?.StatusDescription;
+                    errorMsg = $"Durable Functions trigger status during PulseAsync: {statusCode}: {statusDesc}";

```

Comment:
I'm a little concerned that we might be missing some important information compared to just doing `e.ToString()`. Was there a particular reason why you decided to have different handling for different `WebExceptionStatus` values?

Reply:
Yeah I'm open to changing this error handling. This if/else was copied straight from the durable ScaleController logic, so my intent was to make the error codes here look as similar to ScaleController errors as possible.

Reply:
I don't recall exactly why the scale controller did it this way, but I would feel better if we just did `e.ToString()`.